### DISCUSSION TITLE
Milvus Entity Count Issue

### DISCUSSION THREAD

**Discussion by jits-dd** [milvus-io/milvus#28435] [Q&A and General discussion] [answered]
Hi Team, I have trying out below code. Which was deleting the deleting the pk if already present and insert the updated value of it. But when i run this script containing this method, my entity count kept on increasing despite of having the deleting the already existing entity and reinserting the updated one with same pk (primary key).  I have also attached the screenshot of my collection after running the script for 1st time and the second time. Note - I made sure that no new entiity was inserted, previously inserted entity was run in the second time also. Please suggest some steps to make this workable, because of this i think similarity and other features are not working because it is creating duplication.

def insert_buyer_embedding(self, buyer_id, full_description, is_insert):
            embeddings = self.get_openai_embedding(full_description)
             buyer_collection = self.vector_dao.get_buyer_collection()
            res = buyer_collection.query(output_fields=["pk"],expr=f"pk == {buyer_id}")
            print("res of buyer embedding is", res)
            if res:
                buyer_collection.delete(expr=f"pk == {buyer_id}")

            buyer_collection.insert([
                {"pk": buyer_id, "buyer_sentence":full_description,
                 "embeddings": embeddings}
            ])

            buyer_collection.flush()
            buyer_collection.load()
            return embeddings
// Collection Definition and its fileds

def create_buyer_collection(self):

        fields = [
            FieldSchema(name="pk", dtype=DataType.INT64,
                        is_primary=True, auto_id=False),
            FieldSchema(name="buyer_sentence", dtype=DataType.VARCHAR,
                        max_length=10000),  # Specify max_length
            FieldSchema(name="embeddings", dtype=DataType.FLOAT_VECTOR, dim=1536)
        ]
        schema = CollectionSchema(fields, "Convert Buyer Data to vector")
        buyer_collection = Collection("Buyer_Vector_New", schema)

        index = {
            "index_type": "IVF_FLAT",
            "metric_type": "COSINE",
            "params": {"nlist": 128},
        }
        buyer_collection.create_index("embeddings", index)
        return buyer_collection
        
        
<img width="1246" alt="Screenshot_2023-11-14_at_10 23 05_PM" src="https://github.com/milvus-io/milvus/assets/135798179/acdd9450-1147-4732-87fa-8fb626850532">
<img width="1247" alt="Screenshot_2023-11-14_at_10 34 51_PM" src="https://github.com/milvus-io/milvus/assets/135798179/04cb7732-8f0f-4c83-ad57-b9c03b7c8993">

**Comment by yhmo** [comment, marked as answer]
I saw your question on the Discord.

There are some problems in your script:
- The "res" returned by collection.query() is always Not Null. So, the "if res:" condition is always True.
- And, no need to call flush() frequently. Usually, we only call flush() one time after all the data is inserted into a collection.
- Also, no need to call load() every time you insert an entity. Only call it one time after you create the collection.
- And, to ensure the previous delete/insert actions are done(so that the query can see the data change), use consistency_level="Strong" for the query() method.

I think the correct code logic should be:
```
def insert_buyer_embedding(self, buyer_id, full_description, is_insert):
            embeddings = self.get_openai_embedding(full_description)
            buyer_collection = self.vector_dao.get_buyer_collection()

            # use consistency_level="Strong" to ensure the previous insert/delete actions are done, it will cost 200ms~400ms to execute the query()
            res = buyer_collection.query(output_fields=["pk"],expr=f"pk == {buyer_id}", consistency_level="Strong")
            print("res of buyer embedding is", res)

            # the query() returns a list, if the list is not empty, call delete() to delete the existing entity
            if len(res) > 0:
                buyer_collection.delete(expr=f"pk == {buyer_id}")

            # insert or update the entity
            buyer_collection.insert([
                {"pk": buyer_id, "buyer_sentence":full_description,
                 "embeddings": embeddings}
            ])

            return embeddings
```
